# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/DrawTargetWebgl.h
# Commit: 9cef48a4e5b6
# Full Hash: 9cef48a4e5b61ad9c4e633b8cfaa005a9220672a
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-09-12 21:40:55
# Regressor Bug: 1789849
# File Overlap Count: 2
# Description:
#   Bug 1789849 - Avoid regenerating clip mask unless necessary. r=aosmond
#   
#   Sometimes the clip state is thrashed when we need to temporarily override
#   clipping to disable it. However, in this case, the clip mask itself remains
#   unchanged. The current invalidation scheme doesn't discern between generation
# ==============================================================================

diff -r aeb32fa2f7d5 -r 9cef48a4e5b6 dom/canvas/DrawTargetWebgl.h
--- a/dom/canvas/DrawTargetWebgl.h	Mon Sep 12 15:01:19 2022 +0000
+++ b/dom/canvas/DrawTargetWebgl.h	Mon Sep 12 15:26:02 2022 +0000
@@ -82,7 +82,11 @@
   // Whether or not the WebGL context has valid contents and is being drawn to
   bool mWebglValid = true;
   // Whether or not the clip state has changed since last used by SharedContext.
-  bool mClipDirty = true;
+  bool mClipChanged = true;
+  // Whether or not the clip state needs to be refreshed. Sometimes the clip
+  // state may be overwritten and require a refresh later, even though it has
+  // not changed.
+  bool mRefreshClipState = true;
   // The framebuffer has been modified and should be copied to the swap chain.
   bool mNeedsPresent = true;
   // The number of layers currently pushed.
